extensions/sql-database-projects/src/test/projectController.test.ts

```diff
@@ -247,26 +247,27 @@ describe('ProjectsController: project controller operations', function (): void
 
 			let publishDialog = TypeMoq.Mock.ofType(PublishDatabaseDialog, undefined, undefined, new ApiWrapper(), proj);
 			publishDialog.callBase = true;
-			publishDialog.setup(x => x.getConnectionUri()).returns(async () => 'fake|connection|uri');
+			publishDialog.setup(x => x.getConnectionUri()).returns(() => new Promise((resolve) => resolve('fake|connection|uri')));
 
 			let projController = TypeMoq.Mock.ofType(ProjectsController);
 			projController.callBase = true;
 			projController.setup(x => x.getPublishDialog(TypeMoq.It.isAny())).returns(() => publishDialog.object);
-			projController.setup(x => x.executionCallback(TypeMoq.It.isAny(), TypeMoq.It.is((_): _ is IPublishSettings => true))).returns(async () => {
+			projController.setup(x => x.executionCallback(TypeMoq.It.isAny(), TypeMoq.It.is((_): _ is IPublishSettings => true))).returns(() => {
 				holler = publishHoller;
-				return undefined;
+				return new Promise((resolve) => resolve(undefined));
 			});
-			projController.setup(x => x.readPublishProfile(TypeMoq.It.isAny())).returns(async () => {
+			projController.setup(x => x.readPublishProfile(TypeMoq.It.isAny())).returns(() => {
 				holler = profileHoller;
-				return {
-					databaseName: '',
-					sqlCmdVariables: {}
-				};
+				return new Promise((resolve) =>
+					resolve({
+						databaseName: '',
+						sqlCmdVariables: {}
+					}));
 			});
 
-			projController.setup(x => x.executionCallback(TypeMoq.It.isAny(), TypeMoq.It.is((_): _ is IGenerateScriptSettings => true))).returns(async () => {
+			projController.setup(x => x.executionCallback(TypeMoq.It.isAny(), TypeMoq.It.is((_): _ is IGenerateScriptSettings => true))).returns(() => {
 				holler = generateHoller;
-				return undefined;
+				return new Promise((resolve) => resolve(undefined));
```

Comment:
`Promise.resolve(undefined)` is the same thing just more concise, right?

Reply:
oh right that is much shorter. Updated to do that